Implemented Responsive SideBar

ghareeb97 opened the pull request:
- Adjusted sidebar styles for mobile/desktop
- Improved sidebar item spacing for mobile
- Ensured full-height sidebar on desktop
- Added mobile toggle button
- added search bar

ghareeb97 committed:
Implemented Responsive SideBar

Aamr-mohamed commented on src/Components/SideBar/SideBar.jsx:
```diff
@@ -1,11 +1,58 @@
+        </li>
+        <li className="sidebar-item">
+          <XCircleIcon className="sidebar-icon" />
+          <a href="/logout">Logout</a>
```
change to useNavigate by react-router-dom (<div onClick={()=>navigate("/somewhere")) better that <a>

Aamr-mohamed commented on src/Components/Dashboard/Dashboard.jsx:
change the dashboard to homepage located in the pages ("/")

Aamr-mohamed commented on src/Components/Dashboard/Dashboard.jsx:
```diff
@@ -0,0 +1,15 @@
+import SideBar from "../SideBar/SideBar";
+
+function Dashboard() {
+  const userString = localStorage.getItem("user");
```
can be done in one line
const user=JSON.parse(localStorage....)

ghareeb97 committed:
Created Dashboard

- Implemented UserCards component to display user information with icon, heading, and paragraph props.
- Populated UserTable component using dummyjson API, including edit and delete buttons.
- Delete functionality removes user from table but doesn't affect server (returns after refresh).
- Edit button redirects user to edit page to modify first and last name (though only displays success toast without actual changes).

ghareeb97 committed:
user name is clickable

redirects to /profile/{user.id}

Aamr-mohamed committed:
Merge branch 'main' into dashboard
